### PR TITLE
Segments types & missing Segment List  method

### PR DESCRIPTION
**Pull request by pannga**

### CODE REVIEW

File: packages/sdk/src/Segments.ts [hunk 1]
```diff
@@ -7,19 +9,28 @@ export class Segments {
...
 		return this.client.delete(`/segments/${encode(segmentId)}`)
 	}
+
+	// I see that list method is missing
+
+	/**
+	 * @see https://docs.voucherify.io/reference?utm_source=github&utm_medium=sdk&utm_campaign=acq#list-segments
+	 */
+	public list(customerId: string) {
+		return this.client.delete(`/customers/${encode(customerId)}/segments`)
```
Comment (darekg11) [inline packages/sdk/src/Segments.ts:34]: Why `list` method is doing `this.client.delete`? It's missing definition for return type

File: packages/sdk/src/types/Segments.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		[filter_conditions: string]: $FixMe
+	}
+	type?: string
+	object?: 'segment'
+}
+
+export type SegmentsGetResponse = SegmentsCreateResponse
+
+export interface SegmentsListResponse {
+	object?: 'list'
```
Comment (darekg11) [inline packages/sdk/src/types/Segments.ts:27]: Drop `?`

File: packages/sdk/src/types/Segments.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	total?: number
```
Comment (darekg11) [inline packages/sdk/src/types/Segments.ts:28]: Drop `?`

File: packages/sdk/src/types/Segments.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	data_ref?: 'data'
```
Comment (darekg11) [inline packages/sdk/src/types/Segments.ts:29]: Drop `?`

File: packages/sdk/src/types/Segments.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	data?: {
+		id?: string
```
Comment (darekg11) [inline packages/sdk/src/types/Segments.ts:31]: Drop `?`

File: packages/sdk/src/types/Segments.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+		name?: string
```
Comment (darekg11) [inline packages/sdk/src/types/Segments.ts:32]: Drop `?`

File: packages/sdk/src/types/Segments.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+		object?: 'segment'
```
Comment (darekg11) [inline packages/sdk/src/types/Segments.ts:33]: Drop `?`

File: packages/sdk/src/types/Segments.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	object?: 'segment'
+}
+
+export type SegmentsGetResponse = SegmentsCreateResponse
+
+export interface SegmentsListResponse {
+	object?: 'list'
+	total?: number
+	data_ref?: 'data'
+	data?: {
```
Comment (darekg11) [inline packages/sdk/src/types/Segments.ts:30]: Drop `?`

**Review by darekg11 — CHANGES_REQUESTED** [review]
Changes requested:
- merge conflicts
- logic error
- types definitions

File: packages/sdk/src/Segments.ts (hunk 1 above)
Comment (pannga) [reply packages/sdk/src/Segments.ts:34]: My bad, sorry.

**Review by darekg11 — CHANGES_REQUESTED** [review]
Please fix merge conflicts and we are going to merge it 🙏

**Review by darekg11 — APPROVED** [review]

**Pull request merged**